File: app/src/main/java/com/artmurka/artmurkaapp/Model/Modules/OrdersRequests.java
package com.artmurka.artmurkaapp.Model.Modules;

import com.artmurka.artmurkaapp.Model.InterfacesModel.IOrderRequest;
import com.artmurka.artmurkaapp.Model.Pojo.ItemList.Orders.Orders;

import java.util.HashMap;

import retrofit2.Call;


public class OrdersRequests implements IOrderRequest {

    @Override
    public Call<Orders> getOrders() {
        UcozApiModule ucoz = new UcozApiModule();
        HashMap<String, String> mapForUcozModule = new HashMap<String, String>();
        mapForUcozModule.put("sort", "id");
        mapForUcozModule.put("order", "desc");
        HashMap<String, String> confForRequest = ucoz.get("GET","uapi/shop/invoices/" ,mapForUcozModule);
        return ApiModule.getClient().getInvoises(confForRequest);
    }
}

File: app/src/main/java/com/artmurka/artmurkaapp/data/model/modules/OrdersRequests.kt
package com.artmurka.artmurkaapp.data.model.modules

import com.artmurka.artmurkaapp.data.model.interfacesmodel.IOrderRequest
import com.artmurka.artmurkaapp.data.model.pojo.itemlist.orders.Orders
import com.artmurka.artmurkaapp.data.model.retrofit.ApiRetrofit

import java.util.HashMap

import retrofit2.Call
import javax.inject.Inject


class OrdersRequests @Inject constructor(val apiModule: ApiRetrofit, val ucoz : UcozApiModule ) : IOrderRequest {

    override val orders: Call<Orders>
        get() {
            val mapForUcozModule = HashMap<String, String>()
            mapForUcozModule["sort"] = "id"
            mapForUcozModule["order"] = "desc"
            val confForRequest = ucoz.get("GET", "uapi/shop/invoices/", mapForUcozModule)
            return apiModule.getInvoises(confForRequest)
        }
}
